src/dnx/dnx.cpp

```diff
@@ -135,35 +137,34 @@ void AppendAppbaseFromFile(const dnx::char_t* path, std::vector<const dnx::char_
     }
 }
 
-void ExpandArgument(const dnx::char_t* value, std::vector<const dnx::char_t*>& expanded_args)
+void ExpandProject(const dnx::char_t* project_path, std::vector<const dnx::char_t*>& expanded_args)
 {
-    if (string_ends_with_ignore_case(value, _X(".dll")) || string_ends_with_ignore_case(value, _X(".exe")))
-    {
-        // "dnx /path/App.dll arg1" --> "dnx --appbase /path/ /path/App.dll arg1"
-        // "dnx /path/App.exe arg1" --> "dnx --appbase /path/ /path/App.exe arg1"
-        // "dnx App.exe arg1" --> "dnx --appbase . App.exe arg1"
-        AppendAppbaseFromFile(value, expanded_args);
-        expanded_args.push_back(allocate_and_copy(value));
+    auto split_idx = split_path(project_path);
 
-        return;
-    }
-
-    if (strings_equal_ignore_case(value, _X(".")))
+    // note that we split the path first and check the file name to handle paths like c:\MyApp\my_project.json
+    if (strings_equal_ignore_case(project_path + (split_idx < 0 ? 0 : split_idx + 1), _X("project.json")))
```

Comment:
`project_path + (split_idx < 0 ? 0 : split_idx + 1)` could be just `project_path + split_idx + 1` but requires a comment